docs/style-guide.rst

```diff
@@ -0,0 +1,75 @@
+Style Guide
```

Comment:
is style guide understood by our users or does it sound too intrusive? given that we know that there are fears that we want to change the way they do analysis (we don't)